src/line.js

```diff
@@ -44,22 +51,91 @@ define([
   };
 
   Line.prototype.addSourceGroup = function(sourceGroup) {
-    this._sources.push(sourceGroup);
+    var newSource = {
```

Comment:
@SCourthial it's a doubly linked list.

Reply:
Tat's possible to find the ready-to-use doubly linked list on the Internet on each language. [Javascript](https://github.com/jasonsjones/doubly-linked-list), for example. So, maybe integrating it can ease the development and make the code clearer and easier.

Also, I think we can create another class on top of DoublyLinkedList, like maybe, sourceGroupDoublyList and encapsulate there the logic for finding collisions or something like that.

I don't consist on it, but if we waste, let's say, 2-3 hours on redeveloping the code and save 5-6 hours in the future, then we can do that easily.

So, the questions are:
1. Is it easy and fast enough?
2. Can it ease the development in the future and save some time?

Reply:
Yes, implementing it would make the code cleaner. I think it could indeed be useful for future tasks.

Reply:
Actually, it would make the code cleaner but I'm not sure it will make us save much time in the near future. Encapsulating the logic is always a good idea but I'm not sure it's worth it now that it's done :/

Reply:
ok, so, let's leave it as it is.